docs/auth/vmware-cloud/provider.md

```diff
@@ -0,0 +1,250 @@
+   docs for details).
+1. Click **Create App**, choose 'Web/Mobile app' and click **Continue**.
+1. Use default settings except:
+   - `App Name` and `App Description` of your choosing.
+   - `Redirect URIs`: `${baseUrl}/api/auth/vmwareCloudServices/handler/frame`
+     where `baseUrl` is the URL where your Backstage backend can be reached;
+     note that VMware Cloud does not support the combination of an `http://`
+     scheme and a `localhost` hostname, so when testing locally it may help to
+     set your backend base URL to `http://127.0.0.1:7007`.
+   - `Refresh Token`: check `Issue refresh token`; refresh tokens are required
+     to prevent forcing users to re-login when they refresh their browser.
+   - `Define Scopes`: check `OpenID` at the bottom.
+1. Click **Create**.
+1. Take note of the `App ID` in the resulting modal; this is the client ID to be
+   used by Backstage.
+
+## Install the provider in the backend
+
+### New backend system
+
+Apps using the [new backend system](../../backend-system/index.md),
+can enable the VMware Cloud provider with a small modification like:
+
+```ts title="packages/backend-next/src/index.ts"
+import { createBackend } from '@backstage/backend-defaults';
+
+const backend = createBackend();
+backend.add(import('@backstage/plugin-auth-backend'));
+/* highlight-add-start */
+backend.add(
+  import('@backstage/plugin-auth-backend-module-vmware-cloud-provider'),
+);
+/* highlight-add-end */
+backend.start();
+```
+
+### Old backend system
+
+This provider was added after the migration of the auth-backend plugin to the
+new backend system, so no default provider factory was added. Because of this,
+the installation procedure for old-style backends is slightly more involved:
+
+```ts title="packages/backend/src/plugins/auth.ts"
+import {
+  DEFAULT_NAMESPACE,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import {
+  createRouter,
+  providers,
+  defaultAuthProviderFactories,
+} from '@backstage/plugin-auth-backend';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
+/* highlight-add-start */
+import {
+  commonSignInResolvers,
+  createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import {
+  vmwareCloudAuthenticator,
+} from '@backstage/plugin-auth-backend-module-vmware-cloud-provider';
+/* highlight-add-end */
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  return await createRouter({
+    logger: env.logger,
+    config: env.config,
+    database: env.database,
+    discovery: env.discovery,
+    tokenManager: env.tokenManager,
+    providerFactories: {
+      ...defaultAuthProviderFactories,
+      /* highlight-add-start */
+      vmwareCloudServices: createOAuthProviderFactory({
+        authenticator: vmwareCloudAuthenticator,
+        signInResolver:
+          commonSignInResolvers.emailLocalPartMatchingUserEntityName(),
+      }),
```

Comment:
We can add the provider as one of the built-in ones for the old system if you want, but tbh I don't think it'd be too useful at this point? Depends a bit on if it's used for access delegation too

Reply:
I'm cool without adding the factory to the core, not a big deal.